xapian-core/weight/weight.cc

```diff
@@ -134,4 +138,25 @@ Weight::unserialise(const string &) const
     throw Xapian::UnimplementedError("unserialise() not supported for this Xapian::Weight subclass");
 }
 
+const Weight *
+Weight::create(const string & s, const Registry & reg)
+{
+    const char *p = s.c_str();
+    std::string scheme = "";
```

Comment:
As I noted in the other PR, it's better not to explicitly initialise `std::string` like this.